### PR TITLE
Sync pending accounts when coming back online

### PR DESCRIPTION
**Pull request by jackstenglein**
Adds a function to the AccountDao to get the list of all pending accounts and a function to the AccountRepository to sync pending accounts with the server. Registers a NetworkCallback on application startup that triggers when the default network becomes available. At that point, the pending accounts are synced with the server.

### CODE REVIEW

File: MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java [hunk 1]
```diff
@@ -116,4 +116,24 @@ void setIsCurrent(String name, boolean isCurrent) {
...
+     */
+    public void syncPendingAccounts() {
+        executor.execute(() -> {
+            List<Account> pendingAccounts = accountDao.getPendingAccounts();
+            if (pendingAccounts.isEmpty()) {
+                return;
+            }
+
+            ApiResponse<List<Account>> response = webService.syncPendingAccounts(pendingAccounts);
+            if (response.isSuccessful()) {
```
Comment (TianyiMu) [inline MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java:132]: Wonder if failures need to trigger some kind of retry?

Currently I see this is only triggered when network becomes available. If the sync after the network comes available fails, does that next sync happens at the next network disconnect/connect?

Something you can consider for scheduling periodic work is WorkManager (https://developer.android.com/topic/libraries/architecture/workmanager). Just a suggestion, I am sure you can do it in other ways.

Comment (jackstenglein) [reply MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java:132]: Yes, if the sync fails, then the next sync will happen at the next network reconnect or when the app is closed and reopened.

Comment (TianyiMu) [reply MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java:132]: I see. I feel next app open is fine for now.

If you want to improve this later, maybe add a TODO here? And once you add the sync for other things, the scheduling sync or retry can be added for all of them? What do you think.

Comment (jackstenglein) [reply MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java:132]: Sure, that sounds like a good plan.

Comment (TianyiMu) [reply MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java:132]: Feel free to add an issue for tracking purpose.

**Pull request merged**